judgments/views/index.py

```diff
@@ -3,22 +3,34 @@
     search_judgments_and_parse_response,
 )
 from caselawclient.search_parameters import SearchParameters
+from caselawclient.responses.search_response import SearchResponse
 from django.http import Http404
 from django.views.generic import TemplateView
 from ds_caselaw_utils import courts as all_courts
+from time import time
+from functools import lru_cache
 
 from judgments.forms import AdvancedSearchForm
 from judgments.utils import api_client
 
 
+@lru_cache(maxsize=4)
```

Comment:
good value of maxsize, we need to remember this for the future in case we have more variants...